src/python/pants/backend/java/dependency_inference/java_parser_launcher.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import logging
+
+import pkg_resources
+
+from pants.backend.java.compile.javac import CompiledClassfiles
+from pants.backend.java.compile.javac_binary import JavacBinary
+from pants.engine.fs import CreateDigest, Digest, FileContent, MergeDigests, RemovePrefix
+from pants.engine.process import BashBinary, Process, ProcessResult
+from pants.engine.rules import Get, collect_rules, rule
+from pants.jvm.resolve.coursier_fetch import (
+    ArtifactRequirements,
+    Coordinate,
+    MaterializedClasspath,
+    MaterializedClasspathRequest,
+)
+from pants.util.logging import LogLevel
+
+logger = logging.getLogger(__name__)
+
+
+_LAUNCHER_BASENAME = "PantsJavaParserLauncher.java"
+
+
+def _load_javaparser_launcher_source() -> bytes:
+    return pkg_resources.resource_string(__name__, _LAUNCHER_BASENAME)
+
+
+def java_parser_artifact_requirements() -> ArtifactRequirements:
+    return ArtifactRequirements.create_from_maven_coordinates_fields(
```

Comment:
I plan on getting rid of `create_from_maven_coordinates_fields`. This could just be `ArtifactRequirements([Coordinate(...), Coordinate(...)])`.

Reply:
Done